src/Avalonia.Base/AvaloniaPropertyRegistry.cs

```diff
@@ -38,6 +41,44 @@ public class AvaloniaPropertyRegistry
         /// </summary>
         internal IReadOnlyCollection<AvaloniaProperty> Properties => _properties.Values;
 
+        private ManualResetEvent _unregisteringManualResetEvent = new ManualResetEvent(true);
+        /// <summary>
+        /// Unregister all<see cref="AvaloniaProperty"/>s registered on types
+        /// </summary>
+        /// <param name="types"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public void UnregisterByModule(IEnumerable<Type> types)
+        {
+            _ = types ?? throw new ArgumentNullException(nameof(types));
+            
+            _unregisteringManualResetEvent.Reset();
+            foreach (var type in types)
```

Comment:
Wrap into try/finally, otherwise it's possible to lock up the entire app if something happens here.